microprofile/health-smallrye/src/main/java/org/wildfly/extension/microprofile/health/deployment/DeploymentProcessor.java

```diff
@@ -22,28 +22,44 @@
 
 package org.wildfly.extension.microprofile.health.deployment;
 
+import static org.jboss.as.weld.Capabilities.WELD_CAPABILITY_NAME;
+
 import io.smallrye.health.SmallRyeHealthReporter;
-import org.jboss.as.ee.weld.WeldDeploymentMarker;
+
+import org.jboss.as.controller.capability.CapabilityServiceSupport;
+import org.jboss.as.server.deployment.Attachments;
 import org.jboss.as.server.deployment.DeploymentPhaseContext;
 import org.jboss.as.server.deployment.DeploymentUnit;
+import org.jboss.as.server.deployment.DeploymentUnitProcessingException;
 import org.jboss.as.server.deployment.DeploymentUnitProcessor;
-import org.jboss.as.weld.deployment.WeldPortableExtensions;
+import org.jboss.as.weld.WeldCapability;
 import org.wildfly.extension.microprofile.health.MicroProfileHealthSubsystemDefinition;
 
 /**
  */
 public class DeploymentProcessor implements DeploymentUnitProcessor {
 
     @Override
-    public void deploy(DeploymentPhaseContext phaseContext) {
-        DeploymentUnit deploymentUnit = phaseContext.getDeploymentUnit();
-        if (WeldDeploymentMarker.isPartOfWeldDeployment(deploymentUnit)) {
-            final SmallRyeHealthReporter healthReporter = (SmallRyeHealthReporter) phaseContext.getServiceRegistry().getService(MicroProfileHealthSubsystemDefinition.HEALTH_REPORTER_SERVICE).getValue();
+    public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitProcessingException {
+        final DeploymentUnit deploymentUnit = phaseContext.getDeploymentUnit();
+        final CapabilityServiceSupport support = deploymentUnit.getAttachment(Attachments.CAPABILITY_SERVICE_SUPPORT);
 
-            WeldPortableExtensions extensions = WeldPortableExtensions.getPortableExtensions(deploymentUnit);
-            extensions.registerExtensionInstance(new CDIExtension(healthReporter), deploymentUnit);
+        final WeldCapability weldCapability;
+        try {
+            weldCapability = support.getCapabilityRuntimeAPI(WELD_CAPABILITY_NAME, WeldCapability.class);
+        } catch (CapabilityServiceSupport.NoSuchCapabilityException e) {
+            //We should not be here since the subsystem depends on weld capability. Just in case ...
+            String msg = String.format("Deployment %s requires use of the '%s' capability but it is not currently registered",
+                    phaseContext.getDeploymentUnit().getName(),
+                    WELD_CAPABILITY_NAME);
```

Comment:
This needs to be in an i18n Logger method.

Reply:
Done